src/lib/draw/polygon_tool.ts

```diff
@@ -16,15 +23,19 @@ import { colors, circleRadius } from "../../colors";
 
 const source = "edit-polygon-mode";
 
+// Properties are guaranteed to exist
+type FeatureWithProps<G extends Geometry> = Feature<G> & {
+  properties: { [name: string]: any };
+};
+
 export class PolygonTool {
   map: Map;
   active: boolean;
-  eventListeners: ((f: Feature<Polygon>) => void)[];
+  eventListeners: ((f: FeatureWithProps<Polygon>) => void)[];
   points: Position[];
   cursor: Feature<Point> | null;
-  hoverPolyon: boolean;
-  hoverPoint: number | null;
-  dragging: boolean;
+  // The number is an index into points
```

Comment:
I collapsed a bunch of state here. One field to say what we're hovering on, and to say if we're dragging